video/out/wayland_common.c

```diff
@@ -1776,6 +1792,109 @@ static char **get_displays_spanned(struct vo_wayland_state *wl)
     return names;
 }
 
+static void get_gpu_drm_formats(struct vo_wayland_state *wl)
```

Comment:
I don't know how I feel about this function living in wayland_common, as it has very little to do with wayland. and we're also adding drm headers to wayland_common.

Also is this even necessary anymore after the 3rd commit in this series? We should be getting the list of supported formats and modifiers from the wayland protocol as well

Reply:
It's completely necessary. Without this, I get broken colors again. What the compositor supports is not the same as what your actual GPU supports.

>I don't know how I feel about this function living in wayland_common, as it has very little to do with wayland. and we're also adding drm headers to wayland_common.

That is fair but it does depend directly on a wayland event. It would be viable to move it to drm_common instead if people wanted but wayland code would call it anyway and probably would be the only thing that would ever call it realistically.

Reply:
I guess it can stay here but we really should look into splitting up wayland_common.c in the future, wayland can be its own subdirectory honestly